cmd/finch/virtual_machine_settings.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/runfinch/finch/pkg/config"
+	"github.com/runfinch/finch/pkg/flog"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newSettingsVMCommand(
+	logger flog.Logger,
+	lca config.LimaConfigApplier,
+	fs afero.Fs,
+	stdout io.Writer,
+) *cobra.Command {
+	settingsVMCommand := &cobra.Command{
+		Use:   "settings",
+		Short: "Configure the virtual machine instance",
+		RunE:  newSettingsVMAction(logger, lca, fs, stdout).runAdapter,
+	}
+
+	settingsVMCommand.Flags().Int("cpus", 0, "the amount of vCPU to dedicate to the virtual machine (restart the vm when applying this change.)")
+	settingsVMCommand.Flags().String("memory", "", "the amount of memory to dedicate to the virtual machine (restart the vm when applying this change.)")
+
+	return settingsVMCommand
+}
+
+type settingsVMAction struct {
+	logger            flog.Logger
+	limaConfigApplier config.LimaConfigApplier
+	fs                afero.Fs
+	stdout            io.Writer
+}
+
+func newSettingsVMAction(
+	logger flog.Logger,
+	lca config.LimaConfigApplier,
+	fs afero.Fs,
+	stdout io.Writer,
+) *settingsVMAction {
+	return &settingsVMAction{
+		logger:            logger,
+		limaConfigApplier: lca,
+		fs:                fs,
+		stdout:            stdout,
+	}
+}
+
+func (sva *settingsVMAction) runAdapter(cmd *cobra.Command, _ []string) error {
+	cpus, err := cmd.Flags().GetInt("cpus")
+	if err != nil {
+		return err
+	}
+
+	memory, err := cmd.Flags().GetString("memory")
+	if err != nil {
+		return err
+	}
+
+	return sva.run(cpus, memory)
```

Comment:
would suggest passing as a struct 
for example 
options {
cpus
memory
}

Reply:
this makes it easier to pass objects around and not having to change interfaces.

Reply:
I agree, so I would modify it to pass a structure object instead of passing the value as is.

Reply:
fixed